patches/tModLoader/Terraria/ModLoader/Default/UnloadedPosIndexing.cs

```diff
@@ -0,0 +1,60 @@
+namespace Terraria.ModLoader.Default
+{
+	internal class UnloadedPosIndexing
+	{
+		public int PosID;
+
+		public UnloadedPosIndexing(int posX, int posY) {
+			this.PosID = posX * Main.maxTilesY + posY; // Order determined in accordance with increasing Y in TileIO ReadModData such that PosID is ordered numerically
+		}
+
+		public void SaveChestInfoToPos(UnloadedChestInfo info) {
+			UnloadedTilesSystem modSystem = ModContent.GetInstance<UnloadedTilesSystem>();
+			var infos = modSystem.chestInfos;
+			int pendingID = infos.IndexOf(info);
+			if (pendingID < 0) {
+				pendingID = 0;
+				while (pendingID < infos.Count && infos[pendingID] != null)
+					pendingID++;
+				if (pendingID == infos.Count)
+					infos.Add(info);
+				else
+					infos[pendingID] = info;
+			}
+			modSystem.chestInfoMap[PosID] = pendingID;
+		}
+
+		public void SaveTileInfoToPos(UnloadedTileInfo info) {
+			UnloadedTilesSystem modSystem = ModContent.GetInstance<UnloadedTilesSystem>();
+			var infos = modSystem.tileInfos;
+			int pendingID = infos.IndexOf(info);
+			if (pendingID < 0) {
+				pendingID = 0;
+				while (pendingID < infos.Count && infos[pendingID] != null)
+					pendingID++;
+				if (pendingID == infos.Count)
+					infos.Add(info);
+				else
+					infos[pendingID] = info;
+			}
+			modSystem.tileInfoMap[PosID] = pendingID;
+		}
+
+		public void SaveWallInfoToPos(UnloadedWallInfo info) {
+			UnloadedTilesSystem modSystem = ModContent.GetInstance<UnloadedTilesSystem>();
+			var infos = modSystem.wallInfos;
+			int pendingID = infos.IndexOf(info);
+			if (pendingID < 0) {
+				pendingID = 0;
+				while (pendingID < infos.Count && infos[pendingID] != null)
+					pendingID++;
+				if (pendingID == infos.Count)
+					infos.Add(info);
+				else
+					infos[pendingID] = info;
+			}
+			modSystem.wallInfoMap[PosID] = pendingID;
+		}
```

Comment:
These 3 methods can be turned into 1 with the right parameters.

Reply:
Alternatively, a private method to handle all the pendingID stuff. Though, if things could be generalized to use generics, that would be good too. Probably would need some changes to UnloadedTilesSystem.

Reply:
So, I refactored all the info subclasses, as since the original refactoring of that, a lot has changed.
I was able to reduce it down to one info class again which meant I could mostly clean this to one method. This should be 'better', although I can't say I feel is now perfect.

Reply:
Alright, got a burst of inspiration and am happy now with the end version of this file.
Very simple class with one method, and short method too.

Reply:
@Mirsario - are you content with the changes made to this file to clean it up?
If so, I'd like to resolve the change and move towards merging